crates/sallyport/tests/integration_tests/syscall.rs

```diff
@@ -724,28 +724,24 @@ fn open() {
 #[test]
 #[serial]
 fn poll() {
-    let dev_null_0 = dev_null();
-    let dev_null_1 = dev_null();
-    let dev_null_2 = dev_null();
-
-    let dev_null_0_fd = dev_null_0.into_raw_fd();
-    let dev_null_1_fd = dev_null_1.into_raw_fd();
-    let dev_null_2_fd = dev_null_2.into_raw_fd();
-
     run_test(2, [0xff; 16], move |i, platform, handler| {
+        let dev_null_0 = dev_null();
```

Comment:
this should be `dev_null().into_raw_fd()` or otherwise we should use `as_raw_fd()` in the constructors.
The FD might get closed while being used with current implementation, I believe, since files are moved in the constructors and thus go out of scope